### PR TITLE
python: do not refresh interpreters on quickpick open

### PR DESCRIPTION
**Pull request by isabelizimm**
related to #3944 

Keeps the lists of interpreters the same, rather than refreshing only the first time the Select Interpreter QuickPick is opened which makes it seem like the Positron dropdown is out of sync. 

This nudges people towards refreshing manually when they have created a new environment, which I believe to be the desired behavior since Positron will not have to continually watch for new envs (which can be expensive, some discussion https://github.com/posit-dev/positron/issues/2003). I've opened #4269 to track better UI to make refreshing interpreters more obvious.

### QA Notes

Scenario 1:

1. Open Positron
2. Create new pyenv environment via something like `pyenv virtualenv 3.9.4 test-startup`
3. Check: no new env discovered in Positron dropdown, no new env discovered in `Python: Select Interpreter`
4. Refresh envs using button in `Python: Select Interpreter`
5. See env populate in both locations

Scenario 2:

1. Open Positron
2. Create new pyenv environment via something like `pyenv virtualenv 3.9.4 test-startup`
3. Close Positron
4. Open Positron
5. See env populate in both locations

### CODE REVIEW

**Comment by isabelizimm** [comment]
not quite sure what is happening with the venv/debugger tests, they seem to hang forever--I am able to run them locally

**Review by seeM — APPROVED** [review]
Code changes look good! Scenario 1 works as expected locally, but scenario 2 doesn't. I have to manually refresh for it to show up, but that seems fine to me. I used a different env name in scenario 2 vs 1, maybe related?

I think the test failure may be related to a change in this PR since I haven't seen the failure before. There's an error:

```
Error: Failed to initialize Python extension for tests after 3 minutes
    at activatePythonExtensionScript (/Users/runner/work/positron/positron/path with spaces/extensions/positron-python/out/test/index.js:58:16)
    at Object.run (/Users/runner/work/positron/positron/path with spaces/extensions/positron-python/out/test/index.js:85:15)
```

I wonder if that has to do with removing interpreter refresh? Some of the test helpers might be assuming it happens and awaiting something

**Review by seeM — CHANGES_REQUESTED** [review]
Oops, previous review should've been "Request Changes" because of the hanging test.

**Comment by isabelizimm** [comment]
> Scenario 1 works as expected locally, but scenario 2 doesn't. 

Updated to refresh on extension activation; so both scenarios should work now! That was the problem with the tests hanging as well, waiting for the refresh  👍  This does not impact the selection of the interpreter, that will start up and interpreter discovery will continue in the background.

File: extensions/positron-python/src/client/pythonEnvironments/index.ts [hunk 1]
```diff
@@ -72,6 +92,7 @@ export async function activate(api: IDiscoveryAPI, ext: ExtensionState): Promise
      */
     const folders = vscode.workspace.workspaceFolders;
     // Trigger discovery if environment cache is empty.
+
```
Comment (seeM) [inline extensions/positron-python/src/client/pythonEnvironments/index.ts:95]:
```suggestion
```

**Review by seeM — DISMISSED** [review]

**Pull request merged**